src/zkevm_specs/evm/execution/calldatasize.py

```diff
@@ -6,7 +6,6 @@
 
 def calldatasize(instruction: Instruction):
     opcode = instruction.opcode_lookup(True)
-    instruction.constrain_equal(opcode, Opcode.CALLDATASIZE)
```

Comment:
Is the reason why we no longer need this check that we now have the responsible_opcode check?

Reply:
Because this check is already done at instruction step state transition (see [here](https://github.com/appliedzkp/zkevm-specs/blob/master/src/zkevm_specs/evm/instruction.py#L181)).